Guide/form.markdown

```diff
@@ -5,8 +5,346 @@
+
+## Simple Forms
+
+Forms usually begin with a `formFor` expression. This is how a simple form can look like:
+
+```haskell
+renderForm :: Post -> Html
+renderForm post = formFor post [hsx|
+    {textField #title}
+    {textareaField #body}
+    {submitButton}
+|]
+```
+
+Calling this form from inside your HSX code will lead to the following HTML being generated:
+
+```html
+<form method="POST" action="/CreatePost" id="" class="new-form">
+    <div class="form-group" id="form-group-post_title">
+        <label for="post_title">Title</label>
+        <input type="text" name="title" id="post_title" class="form-control">
+    </div>
+
+    <div class="form-group" id="form-group-post_body">
+        <label for="post_body">Body</label>
+        <textarea name="body" id="post_body" class="form-control"></textarea>
+    </div>
+
+    <button class="btn btn-primary">Create Post</button>
+</form>
+```
+
+You can see that the form is submitted via `POST`. The form action has also been set by default to `/CreatePost`.
+
+All inputs have auto-generated class names and ids for styling. Also all `name` attributes are set as expected.
+
+## Form Controls
+
+IHP has the most commonly used form controls built in. In general the form control helpers just need to be passed the field name. Here is a list of all built-in form control helpers:
+
+```html
+{textField #title}
+{textareaField #body}
+{colorField #brandColor}
+{emailField #email}
+{dateField #dueAt}
+{passwordField #password}
+{dateTimeField #createdAt}
+{hiddenField #projectId}
+{checkboxField #termsAccepted}
+{selectField #projectId allProjects}
+{submitButton}
+```
+
+A form control is always filled with the value of the given field when rendering. For example given a post
+
+```haskell
+let post = Post { ..., title = "Hello World" }
+```
+
+Rendering `{textField #title}`, the input value will be set like
+
+```html
+<input ... value="Hello World">
+```
+
+## Validation
+
+When rendering a record which has failed validation, the validation error message will be rendered automatically.
+
+Given a post like this:
+
+
+```haskell
+let post = Post { ..., title = "" }
+    |> validateField #title nonEmpty
+```
+
+Rendering `{textField #title}`, the input will have the css class `is-invalid` and an element with the error message will be rendered below the input:
+
+```html
+<div class="form-group" id="form-group-post_title">
+    <label for="post_title">Title</label>
+    <input type="text" name="title" placeholder="" id="post_title" class="form-control is-invalid ">
+    <div class="invalid-feedback">This field cannot be empty</div>
+</div>
+```
+
+## Forms Are Also HSX
+
+It's important to understand that the while the form helpers like `{textField #title}` are called by `formFor`, you can still use HSX there. So you can just add any kind of HSX code inside your form:
```

Comment:
"It's important to understand that **_the_** while the form helpers ..."

Reply:
Fixed, good catch :)